crypto/ube/fork_detect.c

```diff
@@ -37,13 +36,31 @@ OPENSSL_STATIC_ASSERT(MADV_WIPEONFORK == 18, MADV_WIPEONFORK_is_not_18)
 #define MADV_WIPEONFORK 18
 #endif
 
-DEFINE_STATIC_ONCE(g_fork_detect_once)
-DEFINE_STATIC_MUTEX(g_fork_detect_lock)
-DEFINE_BSS_GET(volatile char *, g_fork_detect_addr)
-DEFINE_BSS_GET(uint64_t, g_fork_generation)
-DEFINE_BSS_GET(int, g_ignore_madv_wipeonfork)
+static CRYPTO_once_t fork_detect_once = CRYPTO_ONCE_INIT;
+static struct CRYPTO_STATIC_MUTEX fork_detect_lock = CRYPTO_STATIC_MUTEX_INIT;
 
-static int init_fork_detect_madv_wipeonfork(void *addr, long page_size) {
+// This pointer is |volatile| because the value pointed to may be changed by
+// external forces (i.e. the kernel wiping the page) thus the compiler must not
+// assume that it has exclusive access to it.
+static volatile char *fork_detect_addr = NULL;
+static uint64_t fork_generation = 0;
+static int ignore_madv_wipeonfork = 0;
+
+static int init_fork_detect_madv_wipeonfork(void **addr_out) {
+
```

Comment:
nit: Though arguably not necessary as the address passed is the address of a value on the stack on line 83.
```suggestion
`GUARD_PTR(addr_out)`
```